internal/expr/typeinfo.go

```diff
@@ -117,6 +137,73 @@ func (mi *mapInfo) getAllMembers() ([]typeMember, error) {
 
 var _ typeInfo = &mapInfo{}
 
+type primitiveTypeInfo struct {
+	primitiveType reflect.Type
+}
+
+func (pti *primitiveTypeInfo) typ() reflect.Type {
+	return pti.primitiveType
+}
+
+func (pti *primitiveTypeInfo) typeMember(member string) (typeMember, error) {
+	if member != "" {
+		return nil, fmt.Errorf(`cannot specify member of primitive type %q`, pti.primitiveType.Name())
+	}
+	return primitiveType{primitiveType: pti.primitiveType}, nil
```

Comment:
Same concern, this error should be moved to the parser and not reported here which would indicate that the type does not conform to the interface, or that the interface is not the correct abstraction.

Reply:
Same comment as above.